### PR TITLE
test: add test for custom covariance and noisy average

### LINKED ISSUE #13: Build Python unit tests (with pytest)
I guess we mainly need tests for our custom implementations, not for the wrapped PyDP, given that this is unit testing.

### PR DESCRIPTION
**Pull request by LaRiffle**
## Fixes

Fixes #13

## Description

I added a basic test for the custom covariance, mainly the setup the basic test workflow. Happy to have some feedback to improve it!

We miss GH action to test all of this, but so far it passes locally 👼 

## Definition of Done

- [x] I followed the [Arkhn Code Book](https://www.notion.so/arkhn/Arkhn-Code-Book-23ac18a8af7343d0a3f61f1c9ef7400c) (I swear!).
- [x] I have written tests for the code I added or updated.
- [ ] I have updated the documentation according to my changes.
- [ ] I have updated the deployment configuration if needed.

### CODE REVIEW

File: tests/unit/dp/custom_dp/covariance_custom_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    n_noisy = n
+
+    x_min = 150
+    x_max = 200
+
+    epsilon = 100
+
+    x_noisy_average = noisy_average(x, epsilon, x_min, x_max, n, n_noisy)
+    x_average = np.mean(x)
+    assert np.abs(x_noisy_average - x_average) / x_average < 0.01
```
Comment (naudinlo) [inline tests/unit/dp/custom_dp/covariance_custom_test.py:19]: What do you think about putting 0.1 in a global variable and create a util function to compute difference between true and noisy computations? 
It would be more meaningful and could be reused in `test_custom_bounded_covariance` and later tests

File: inst/python/dp/custom_dp/covariance_custom.py [hunk 2]
```diff
@@ -108,9 +107,7 @@ def custom_bounded_covariance(
     xy_normalized = (x - noisy_average_x) * (y - noisy_average_y)
 
     # Compute the noisy average of the normalized product, ie the covariance
-    noisy_average_xy = noisy_average(
-        xy_normalized, eps_per_query, xy_min, xy_max, n, n_noisy
-    )
+    noisy_average_xy = noisy_average(xy_normalized, eps_per_query, xy_min, xy_max, n, n_noisy)
```
Comment (naudinlo) [inline inst/python/dp/custom_dp/covariance_custom.py:110]: As discussed, linter/ syntax validator would be cool to avoid this!

**Review by naudinlo — APPROVED** [review]
Just two remarks:
- We could test edge cases as well 
- I don't think you need that many `__init__` files!

File: inst/python/dp/custom_dp/covariance_custom.py (hunk 2 above)
Comment (LaRiffle) [reply inst/python/dp/custom_dp/covariance_custom.py:110]: Yes! It's part of this work :) 

https://github.com/arkhn/dsPrivacy/pull/14

**Comment by LaRiffle** [comment]
@naudinlo what edge cases would you consider for example?

**Comment by LaRiffle** [comment]
@naudinlo I've added tests for handling empty values, but it is not very satisfactory (see comments in code).
However, I'd be in favor to merge the code as is and open an issue for future work, to be able to add the test structure to the project today
Sounds good to you?
